icu4c/source/i18n/number_formatimpl.cpp

```diff
@@ -36,7 +36,7 @@ int32_t NumberFormatterImpl::formatStatic(const MacroProps &macros, UFormattedNu
     NumberFormatterImpl impl(macros, false, status);
     MicroProps& micros = impl.preProcessUnsafe(inValue, status);
     if (U_FAILURE(status)) { return 0; }
-    int32_t length = writeNumber(micros, inValue, outString, 0, status);
+    int32_t length = writeNumber(micros.toSimple(), inValue, outString, 0, status);
```

Comment:
Will the construction of a temporary SimpleMicroProps degrade the performance of the regular NumberFormatter noticeably?

Reply:
Hmm. Creating the SimpleMicroProps is just copying around pointers. In Rust, to avoid this, I would use a (static) trait bound like `impl AsRef<SimpleMicroProps>` or similar. I guess one approach that would be similar would be to make MicroProps wrap a SimpleMicroProps (as a field) so that I can actually take that reference.

Reply:
Fixed this by making SimpleMicroProps a field of MicroProps.